### ISSUE TITLE
modin is duplicating the index

### ISSUE THREAD

**Issue by jschewebbn** [modin-project/modin#516]
When running the following code you will see that the last line gives back a number of rows of data with duplicate index values.
```python
import modin.pandas as pd

frames = list()
df1 = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))
frames.append(df1)

df2 = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))
frames.append(df2)

df3 = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))
frames.append(df3)

data = pd.concat(frames, ignore_index=True)

data[data.index.duplicated()]
```

If you change the import to "import pandas as pd" and execute again the final line returns zero rows.

**Comment by devin-petersohn** [comment]
Thanks @jschewebbn, I see where the issue is.

`modin.pandas.concat` is incorrectly ignoring the `ignore_index` parameter. It is handled correctly if we use `append`, but those code paths probably need to be merged:

```python
import modin.pandas as pd

frames = list()
df1 = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))

df2 = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))
frames.append(df2)

df3 = pd.DataFrame(np.random.randint(0,100,size=(100, 4)), columns=list('ABCD'))
frames.append(df3)

data = df1.append(frames, ignore_index=True)
```

I will flag this fix for the next release. It is a relatively small change since it is already working in `append`, so I will also flag for Help Wanted until I can fix it myself.

The code you provided has also revealed a bug related to masking with all `False` values (`data[data.index.duplicated()]`). Somehow cloudpickle is trying to pickle an infinitely recursive object. This should be tracked separately, so I will open an issue for that. Thanks!

**Comment by devin-petersohn** [comment]
Closed via #521
